LaunchPadBooster/Utils/InputLock.cs

```diff
@@ -0,0 +1,91 @@
+﻿using Assets.Scripts;
+using Assets.Scripts.UI;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace LaunchPadBooster.Utils
+{
+    /// <summary>
+    /// Creating this object will create an input lock in the game.
+    /// This prevents keyboard or mouse input from doing what it normally does.
+    /// Cancelling the lock is done by calling Dispose() on the object.
+    /// </summary>
+    public class InputLock : IDisposable
```

Comment:
instead of a class you can use a struct here to avoid allocating on the heap just for this